[vulkan_api][ops] Mm, Pool, Upsample

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47063 [vulkan_api][ops] Mm, Pool, Upsample**

Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

IvanKobzarev committed:
[vulkan_api][ops] Mm, Pool, Upsample

[ghstack-poisoned]

IvanKobzarev committed:
cleanup on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

SS-JIA commented on aten/src/ATen/native/vulkan/ops/Mm.cpp:
```diff
@@ -0,0 +1,138 @@
+        mat2_arg.sizes()[1],
+        mat1_arg.sizes()[0],
+        1u,
+        mat1_arg.sizes()[1],
```
This should be the last argument no? (i.e. corresponds to K in shader Block)

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

SS-JIA commented on aten/src/ATen/native/vulkan/ops/Mm.cpp:
```diff
@@ -0,0 +1,139 @@
+
+  vTensor v_output{
+    context,
+    self.sizes(),
```
I changed this to
`{mat1.sizes()[0], mat2.sizes()[1]},`
and the addmm test is now passing on my laptop

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

SS-JIA commented on aten/src/ATen/native/vulkan/ops/Upsample.cpp:
```diff
@@ -0,0 +1,80 @@
+  command_buffer.begin();
+  {
+    const float scale_x = compute_scales_value<float>(scales_w, input_arg.sizes()[3], output_sizes[2]);
+    const float scale_y = compute_scales_value<float>(scales_h, input_arg.sizes()[3], output_sizes[2]);
```
I think the scales are being computed wrong here (mismatched indices), which is causing the test to fail.

Should be something like
`scale_x = input_arg.sizes()[2]/output_sizes[0]`
`scale_y = input_arg.sizes()[2]/output_sizes[1]`

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

SS-JIA commented on aten/src/ATen/native/vulkan/ops/Upsample.cpp:
```diff
@@ -0,0 +1,80 @@
+
+  vTensor v_output{
+    context,
+    output_sizes,
```
Since `output_sizes` only has 2 elements, I also think this should be something like

`{input_arg.sizes()[0], input_arg.sizes()[1], output_sizes[0], output_sizes[1]`

SS-JIA commented on aten/src/ATen/native/vulkan/ops/Upsample.cpp:
```diff
@@ -0,0 +1,80 @@
+        input_arg.sizes()[3],
+        input_arg.sizes()[2],
+        output_sizes[3],
+        output_sizes[2],
```
Should be `output_sizes[1], output_sizes[0]` here as well

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan_api][ops] Mm, Pool, Upsample"


Differential Revision: [D24624610](https://our.internmc.facebook.com/intern/diff/D24624610)

[ghstack-poisoned]
